src/layers/invertible_layer_conv1x1.jl

```diff
@@ -110,7 +110,10 @@ end
 function mat_tens_i(out::AbstractVector{T}, Mat::AbstractArray{T, 2},
                     Tens::AbstractArray{T, 3}, Mat2::AbstractArray{T, 2}) where T
     # Computes sum( (Mat * tens) .* Mat2) for each element in the batch
+
+    isa(Mat, CUDA.CuArray) && (Mat2 = CUDA.CuArray(Mat2)) #new Julia 1.10 subarrays require this
```

Comment:
I think just `Mat2 = collect(Mat2)` should always work

Reply:
Darn looks like that puts the Mat2 on CPU:
```    
Mat2 = collect(Mat2)
copyto!(out, map(i -> dot(Mat * Tens[i, :, :], Mat2) , 1:size(Tens, 1)))
```
```
  LoadError: MethodError: no method matching dot(::Int64, ::CUDA.CuPtr{Float32}, ::Int64, ::Ptr{Float32}, ::Int64)
  
  Closest candidates are:
    dot(::Integer, ::Union{Ptr{Float32}, AbstractArray{Float32}}, ::Integer, ::Union{Ptr{Float32}, AbstractArray{Float32}}, ::Integer)
     @ LinearAlgebra /net/coc-herrmann/GATechBundle/Julia/julia-1.10.0/share/julia/stdlib/v1.10/LinearAlgebra/src/blas.jl:344
    dot(::Integer, ::Union{Ptr{Float64}, AbstractArray{Float64}}, ::Integer, ::Union{Ptr{Float64}, AbstractArray{Float64}}, ::Integer)
     @ LinearAlgebra /net/coc-herrmann/GATechBundle/Julia/julia-1.10.0/share/julia/stdlib/v1.10/LinearAlgebra/src/blas.jl:344
  
  Stacktrace:
    [1] dot(x::CUDA.CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, y::Matrix{Float32})
      @ LinearAlgebra.BLAS /net/coc-herrmann/GATechBundle/Julia/julia-1.10.0/share/julia/stdlib/v1.10/LinearAlgebra/src/blas.jl:395
    [2] dot(x::CUDA.CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, y::Matrix{Float32})
      @ LinearAlgebra /net/coc-herrmann/GATechBundle/Julia/julia-1.10.0/share/julia/stdlib/v1.10/LinearAlgebra/src/matmul.jl:15
    [3] (::InvertibleNetworks.var"#249#250"{CUDA.CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, CUDA.CuArray{Float32, 3, CUDA.Mem.DeviceBuffer}, Matrix{Float32}})(i::Int64)
      @ InvertibleNetworks ~/.julia/dev/InvertibleNetworks/src/layers/invertible_layer_conv1x1.jl:116
    [4] iterate
      @ Base ./generator.jl:47 [inlined]
    [5] _collect
      @ Base ./array.jl:854 [inlined]
    [6] collect_similar
      @ Base ./array.jl:763 [inlined]
    [7] map
      @ Base ./abstractarray.jl:3282 [inlined]
    [8] mat_tens_i(out::CUDA.CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}, Mat::CUDA.CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, Tens::CUDA.CuArray{Float32, 3, CUDA.Mem.DeviceBuffer}, Mat2::Base.ReshapedArray{Float32, 2, SubArray{Float32, 3, CUDA.CuArray{Float32, 4, CUDA.Mem.DeviceBuffer}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Tuple{}})
```